Minio/DataModel/BucketOperationsArgs.cs

```diff
@@ -0,0 +1,160 @@
+            Region = "";
+            Secure = false;
+            BucketName = "";
+        }
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+        }
+        public new RestRequest GetRequest(string baseUrl, RestSharp.Authenticators.IAuthenticator authenticator)
+        {
+            return RequestUtil.CreateRequest(baseUrl, Method.HEAD, authenticator, this.BucketName, this.Secure, this.Region );
+        }
+        public new BucketExistsArgs WithBucket(string bucket)
+        {
+            return (BucketExistsArgs)(base.WithBucket(bucket));
+        }
+
+        public new BucketExistsArgs WithLocation(string loc)
+        {
+            return (BucketExistsArgs)(base.WithLocation(loc));
+        }
+        public new BucketExistsArgs WithRegion(string reg)
+        {
+            return (BucketExistsArgs)(base.WithRegion(reg));
+        }
+        public new BucketExistsArgs WithSSL(bool secure=true)
+        {
+            return (BucketExistsArgs)(base.WithSSL(secure));
+        }
+    }
+
+    public class RemoveBucketArgs : BucketArgs
+    {
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+        }
+        public new RemoveBucketArgs WithBucket(string bucket)
+        {
+            return (RemoveBucketArgs)(base.WithBucket(bucket));
+        }
+
+        public new RemoveBucketArgs WithLocation(string loc)
+        {
+            return (RemoveBucketArgs)(base.WithLocation(loc));
+        }
+        public new RemoveBucketArgs WithRegion(string reg)
+        {
+            return (RemoveBucketArgs)(base.WithRegion(reg));
+        }
+        public new RestRequest GetRequest(string baseUrl, RestSharp.Authenticators.IAuthenticator authenticator)
+        {
+            return RequestUtil.CreateRequest(baseUrl, Method.DELETE, authenticator, this.BucketName, this.Secure, this.Region);
+        }
+        public new RemoveBucketArgs WithSSL(bool secure=true)
+        {
+            return (RemoveBucketArgs)(base.WithSSL(secure));
+        }
+    }
+
+    public class ListObjectsArgs : BucketArgs
+    {
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+        }
+        public new ListObjectsArgs WithLocation(string loc)
+        {
+            return (ListObjectsArgs)(base.WithLocation(loc));
+        }
+        public new ListObjectsArgs WithRegion(string reg)
+        {
+            return (ListObjectsArgs)(base.WithRegion(reg));
+        }
+        public new RestRequest GetRequest(string baseUrl, RestSharp.Authenticators.IAuthenticator authenticator)
+        {
+            return RequestUtil.CreateRequest(baseUrl, Method.GET, authenticator, this.BucketName, this.Secure,  this.Region);
+        }
+        public new ListObjectsArgs WithSSL(bool secure=true)
+        {
+            return (ListObjectsArgs)(base.WithSSL(secure));
+        }
+    }
```

Comment:
for bucket operations, only MakeBucket requires location and region. Can you remove the WithLocation and WithRegion methods from all the remaining operations.

Reply:
docs is missing example for ListObjectsAsync with this args variant.

Reply:
can you also change the functional tests and example programs to this new style.

Reply:
ListObjectsAsync isn't implemented yet. Everything else it needs is implemented. I'll implement ListObjectsAsync in the next PR.

Reply:
Some cases (MakeBucket, RemoveBucket, BucketExists and Versioning) are implemented. Also Unit Tests.  Changing functional test(Program.cs).